### DISCUSSION TITLE
How to config a pkuseg tokenizer to a trained ner model?

### DISCUSSION THREAD

**Discussion by baiziyuandyufei** [explosion/spaCy#7177] [Help: Coding & Implementations]
## Your Environment

- **spaCy version:** 3.0.0rc5
- **Platform:** Darwin-20.3.0-x86_64-i386-64bit
- **Python version:** 3.6.12
- **Pipelines:** zh_core_web_trf (3.0.0a1)

use the config file generated by [https://spacy.io/usage/training](https://spacy.io/usage/training), get the model, the tokenizer.segmenter is 'char'.

```
>>> import spacy
>>> nlp = spacy.load("model-best-trf")
>>> nlp.tokenizer.segmenter
'char'
```

but the 'zh_core_web_trf-3.0.0a1' model's segmenter is 'pkuseg', How realize? 

```
>>> import spacy
>>> nlp = spacy.load("zh_core_web_trf")
>>> nlp.tokenizer.segmenter
'pkuseg'
```

I need to modify the training data to per word one line, not per character one line?

```
我                  O
非常              O
喜欢              O
贾                  B-PERSON
思                  I-PERSON
琪                  I-PERSON
。                  O
```

**Comment by adrianeboyd** [comment]
Here are the details on the Chinese options: https://spacy.io/usage/models#chinese

Your config would include blocks like this:

```ini
[nlp.tokenizer]
@tokenizers = "spacy.zh.ChineseTokenizer"
segmenter = "pkuseg"
```

```ini
[initialize.tokenizer]
pkuseg_model = "assets/pkuseg_model"
pkuseg_user_dict = "default"
```

`pkuseg_model` is either a local path to the pkuseg model or one of the supported model names from the original pkuseg package (see the "Details on trained and custom Chinese pipelines" expandable box in the docs above). If you want to use the same model as `zh_core_web_*" model, it's saved in the model package under `tokenizer/pkuseg_model`.

If you have only NER training data, the tokenization is not as important as long as your NER span boundaries fall on token boundaries. (With tags/parses, it's a lot more crucial that each word line up or the model can't train well.) If you have trouble with the NER spans aligning with pkuseg segmentation, try the `char` segmenter instead, which should work fine for training a transformer + NER pipeline.
